rclcpp/test/rclcpp/test_logger.cpp

```diff
@@ -160,6 +160,39 @@ TEST(TestLogger, set_level) {
   EXPECT_EQ(RCUTILS_RET_OK, rcutils_logging_shutdown());
 }
 
+TEST(TestLogger, get_effective_level) {
+  ASSERT_EQ(RCUTILS_RET_OK, rcutils_logging_initialize());
+
+  rclcpp::Logger logger = rclcpp::get_logger("test_logger");
+
+  // default
+  EXPECT_EQ(rclcpp::Logger::Level::Info, logger.get_effective_level());
+
+  // unset
+  logger.set_level(rclcpp::Logger::Level::Unset);
+  EXPECT_EQ(rclcpp::Logger::Level::Info, logger.get_effective_level());
+
+  // debug
+  logger.set_level(rclcpp::Logger::Level::Debug);
+  EXPECT_EQ(rclcpp::Logger::Level::Debug, logger.get_effective_level());
+
+  // info
+  logger.set_level(rclcpp::Logger::Level::Info);
+  EXPECT_EQ(rclcpp::Logger::Level::Info, logger.get_effective_level());
+
+  // warn
+  logger.set_level(rclcpp::Logger::Level::Warn);
+  EXPECT_EQ(rclcpp::Logger::Level::Warn, logger.get_effective_level());
+
+  // error
+  logger.set_level(rclcpp::Logger::Level::Error);
+  EXPECT_EQ(rclcpp::Logger::Level::Error, logger.get_effective_level());
+
+  // fatal
+  logger.set_level(rclcpp::Logger::Level::Fatal);
+  EXPECT_EQ(rclcpp::Logger::Level::Fatal, logger.get_effective_level());
```

Comment:
How about adding some more tests about `rclcpp::get_logger("test_logger.child")`?

Reply:
yeah that makes sense.

Reply:
addressed by https://github.com/ros2/rclcpp/pull/2141/commits/b185c701093ffbe3d711c61fe07794e7875bc6f2